vm/transaction.go

```diff
@@ -23,7 +18,8 @@ func marshalTxn(txn core.Transaction) (json.RawMessage, error) {
 		return nil, err
 	}
 
-	t.Version = clearQueryBit(t.Version)
+	versionWithoutQueryBit := (*core.TransactionVersion)(t.Version).WithoutQueryBit()
+	t.Version = versionWithoutQueryBit.AsFelt()
```

Comment:
Can we do `txn.WithoutQueryBit` before we convert to the feeder type?

Reply:
Sadly no, we dont have access to `Version` field thru `core.Transaction` interface.
Alternative could be doing it in the switch below, but it is far from DRY.